src/components/TodoForm/TodoForm.js

```diff
@@ -0,0 +1,189 @@
+    name: '',
+    userId: 0,
+    title: '',
+    errorTitle: false,
+    errorPersone: false,
+    errorLength: false,
+    errorNotValid: false,
+  }
+
+  handleTitleChange = ({ target }) => {
+    if (target.value.length > 40) {
+      this.setState({
+        title: target.value.slice(0, 40),
+        errorLength: true,
+        errorTitle: false,
+        errorNotValid: false,
+      });
+    } else {
+      this.setState({
+        title: target.value,
+        errorTitle: false,
+        errorLength: false,
+        errorNotValid: false,
+      });
+    }
+
+    if (target.value.match(/[^A-Za-zА-Яа-я ]/)) {
+      this.setState({
+        errorNotValid: true,
+        title: target.value.slice(0, target.value.search(/[^A-Za-zА-Яа-я ]/)),
+      });
+    }
+  }
+
+  personeSelection = ({ target }) => {
+    const user = this.props.users.find(currentUser => (
+      currentUser.name === target.value
+    ));
+
+    this.setState({
+      name: target.value,
+      userId: user.id,
+      errorPersone: false,
+    });
+  }
+
+  handleFormSubmit = (event, title, name, newID) => {
+    event.preventDefault();
+    if (title === '') {
+      this.setState({
+        errorTitle: true,
+      });
+    }
+
+    if (name === '') {
+      this.setState({
+        errorPersone: true,
+      });
+    }
+
+    if (title !== '' && name !== '') {
+      this.setState(state => ({
+        todos: [
+          ...state.todos,
+          {
+            userId: state.userId,
+            id: newID,
+            title: state.title,
+            completed: false,
+          },
+        ],
+        name: '',
+        title: '',
+        errorLength: false,
+        errorNotValid: false,
+      }));
+    }
+  }
+
+  render() {
+    const { name, todos, title, errorTitle, errorPersone,
+      errorLength, errorNotValid } = this.state;
+    const { users } = this.props;
+    const identifiers = todos.map(todo => todo.id);
+    const newID = Math.max(...identifiers) + 1;
+    const preparedTodos = todos.map(todo => (
+      {
+        ...todo,
+        user: users.find(user => todo.userId === user.id),
+      }));
+
+    return (
+      <>
+        <form
+          action="#"
+          method="GET"
+          className="TodoForm"
+          onSubmit={event => this.handleFormSubmit(event, title, name, newID)}
+        >
+          <label
+            className="TodoForm__label"
+            htmlFor="title"
+          >
+            Please enter task to be done:
+          </label>
+          <input
+            name="title"
+            type="text"
+            id="title"
+            value={title}
+            onChange={this.handleTitleChange}
+          />
+          {errorTitle
+          && (
+          <p className="TodoForm__error">
+            *Please enter the title
+          </p>
+          )}
+          {errorLength
+          && (
+          <p className="TodoForm__error">
+            *Maximum length exceeded
+          </p>
```

Comment:
better make error mes hidden, because he will reserves place. In your solution he disappear and appears and changing forms height